packages/cli/src/ui/components/Footer.test.tsx

```diff
@@ -33,50 +44,91 @@ const defaultProps = {
   targetDir:
     '/Users/test/project/foo/bar/and/some/more/directories/to/make/it/long',
   branchName: 'main',
-  debugMode: false,
-  debugMessage: '',
-  corgiMode: false,
-  errorCount: 0,
-  showErrorDetails: false,
-  showMemoryUsage: false,
-  promptTokenCount: 100,
-  nightly: false,
 };
 
-const renderWithWidth = (width: number, props = defaultProps) => {
+const createMockConfig = (overrides = {}) => ({
+  getModel: vi.fn(() => defaultProps.model),
+  getTargetDir: vi.fn(() => defaultProps.targetDir),
+  getDebugMode: vi.fn(() => false),
+  ...overrides,
+});
+
+const createMockUIState = (overrides: Partial<UIState> = {}): UIState =>
+  ({
+    sessionStats: {
+      lastPromptTokenCount: 100,
+    },
+    branchName: defaultProps.branchName,
+    ...overrides,
+  }) as UIState;
+
+const createDefaultSettings = (
+  options: {
+    showMemoryUsage?: boolean;
+    hideCWD?: boolean;
+    hideSandboxStatus?: boolean;
+    hideModelInfo?: boolean;
+  } = {},
+): LoadedSettings =>
+  ({
+    merged: {
+      ui: {
+        showMemoryUsage: options.showMemoryUsage,
+        footer: {
+          hideCWD: options.hideCWD,
+          hideSandboxStatus: options.hideSandboxStatus,
+          hideModelInfo: options.hideModelInfo,
+        },
+      },
+    },
+  }) as never;
+
+const renderWithWidth = (
```

Comment:
This is a great test helper and should be standard test helper rather than something in the footer.
Can you you move this to  test-utils/render.tsx? 

We can leave using it in more places as a follow up.